### ISSUE TITLE
writeMPS and writeLP is slow

### ISSUE THREAD

**Issue by sebastroy** [coin-or/pulp#101]
Hi,

I took a look into the two writeMPS and writeLP functions in pulp.py. I find that the vs = self.variables() and the repeated loops over constraints to be sometimes as slow as running cbc itself.

I have an alternative implementation ready for a pull request but I wanted to hear opinions here first.

I also notice that if there is a lot of unused variables (with bounds set to zero) even with this code shuffling, it gets very long. Same for the lp.assign\* instructions near the end of the CBC_solve function. I had inquired about variables removal in https://groups.google.com/forum/#!topic/pulp-or-discuss/BuLRw_qOuIg and it seems that not removing the variables is a bigger bottleneck than cbc itself.

While I can use to proposed solution and remove variables by "hand" I think it would be cleaner to add an interface for variable removal.

Am I missing something else?

Thanks!

**Comment by RomanZavodskikh** [comment]
Yeah, I'm using PuLP now and I'm stucked on this bottleneck too. You can see Flame Graph illustating my issue here.
![perf](https://user-images.githubusercontent.com/8851784/42264673-010977cc-7f7a-11e8-9e31-26fd5a5f6d3f.jpg)
So, @sebastroy, this problem is actual.
Moreover, you should feel free to open pull requests in such cases. Creation of it won't affect work of PuLP users. This PR will be reviewed by appropriate people and discussed.
So if you're still aware please, open pull request. I will use your changes in my own project. Maybe, they will be used by someone else.

**Comment by sebastroy** [comment]
I did a fork instead because the pulp team is small and PR weren't treated in a timely enough fashion for my needs at the time:

https://github.com/sebastroy/pulp/commit/ee00e5d22bae461c2821ad147c807e152bd62cab

It's been a very long while but I had done it then. Not sure if the commit is still valid on coin-or/pulp master. 

That being said, we switched to or-tools since then. Pulp writes an input file for the "bash" cbc executable and reads it's output file. When your LP grows, we have seen cases where writting and reading the file would be much longer than actually solving the LP. As a result we switched to an in-memory implementation (like or-tools does). I recommend you do the same. There were other missing features from pulp that existed in or-tools that I don't remember the details.

or-tools is sponsored by google and although somewhat slow in addressing issues, there is a full time person dedicated to the projet so it tends to be more useful. Although there are too many features which as far as the LP is concerned is counter producive.
